library/core/src/intrinsics/simd.rs

```diff
@@ -0,0 +1,412 @@
+    ///
+    /// `T` must be a vector of floating-point primitive types.
+    ///
+    /// `U` must be a vector of integers with the same number of elements and element size as `T`.
+    pub fn simd_le<T, U>(x: T, y: T) -> U;
+
+    /// Tests if `x` is greater than `y`, elementwise.
+    ///
+    /// Returns `0` for false and `!0` for true.
+    ///
+    /// `T` must be a vector of floating-point primitive types.
+    ///
+    /// `U` must be a vector of integers with the same number of elements and element size as `T`.
+    pub fn simd_gt<T, U>(x: T, y: T) -> U;
+
+    /// Tests if `x` is greater than or equal to `y`, elementwise.
+    ///
+    /// Returns `0` for false and `!0` for true.
+    ///
+    /// `T` must be a vector of floating-point primitive types.
+    ///
+    /// `U` must be a vector of integers with the same number of elements and element size as `T`.
+    pub fn simd_ge<T, U>(x: T, y: T) -> U;
+
+    /// Shuffle two vectors by const indices.
+    ///
+    /// Concatenates `x` and `y`, then returns a new vector such that each element is selected from
+    /// the concatenation by the matching index in `idx`.
+    ///
+    /// `T` must be a vector.
+    ///
+    /// `U` must be a const array of `i32`s.
+    ///
+    /// `V` must be a vector with the same element type as `T` and the same length as `U`.
+    pub fn simd_shuffle<T, U, V>(x: T, y: T, idx: U) -> V;
+
+    /// Read a vector of pointers.
+    ///
+    /// For each pointer in `ptr`, if the corresponding value in `mask` is `!0`, read the pointer.
+    /// Otherwise if the corresponding value in `mask` is `0`, return the corresponding value from
+    /// `val`.
+    ///
+    /// `T` must be a vector.
+    ///
+    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
+    ///
+    /// `V` must be a vector of integers with the same length as `T` (but any element size).
+    ///
+    /// # Safety
+    /// Unmasked values in `T` must be readable as if by `<ptr>::read` (e.g. aligned to the element
+    /// type).
+    ///
+    /// `mask` must only contain `0` or `!0` values.
+    pub fn simd_gather<T, U, V>(val: T, ptr: U, mask: V) -> T;
+
+    /// Write to a vector of pointers.
+    ///
+    /// For each pointer in `ptr`, if the corresponding value in `mask` is `!0`, write the
+    /// corresponding value in `val` to the pointer.
+    /// Otherwise if the corresponding value in `mask` is `0`, do nothing.
+    ///
+    /// `T` must be a vector.
+    ///
+    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
+    ///
+    /// `V` must be a vector of integers with the same length as `T` (but any element size).
+    ///
+    /// # Safety
+    /// Unmasked values in `T` must be writeable as if by `<ptr>::write` (e.g. aligned to the element
+    /// type).
+    ///
+    /// `mask` must only contain `0` or `!0` values.
+    pub fn simd_scatter<T, U, V>(val: T, ptr: U, mask: V);
+
```

Comment:
```suggestion
    /// Read a vector.
    ///
    /// For each element offset from `ptr`, if the corresponding value in `mask` is `!0`, read the pointer.
    /// Otherwise if the corresponding value in `mask` is `0`, return the corresponding value from
    /// `val`.
    ///
    /// `T` must be a vector.
    ///
    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
    ///
    /// `V` must be a vector of integers with the same length as `T` (but any element size).
    ///
    /// # Safety
    /// Unmasked values in `T` must be readable as if by `<ptr>::read` (e.g. aligned to the element
    /// type).
    ///
    /// `mask` must only contain `0` or `!0` values.
    pub fn simd_masked_load<V, U, T>(mask: V, ptr: U, val: T) -> T;

    /// Write a vector.
    ///
    /// For each element offset from `ptr`, if the corresponding value in `mask` is `!0`, write the
    /// corresponding value in `val` to the pointer.
    /// Otherwise if the corresponding value in `mask` is `0`, do nothing.
    ///
    /// `T` must be a vector.
    ///
    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
    ///
    /// `V` must be a vector of integers with the same length as `T` (but any element size).
    ///
    /// # Safety
    /// Unmasked values in `T` must be writeable as if by `<ptr>::write` (e.g. aligned to the element
    /// type).
    ///
    /// `mask` must only contain `0` or `!0` values.
    pub fn simd_masked_store<V, U, T>(mask: V, ptr: U, val: T);
```

Something like this perhaps?